### ISSUE TITLE
Indentation of empty lines within [], (), and {} inconsistent

### ISSUE THREAD

**Issue by AdamNiederer** [orzechowskid/tsi.el#23]
`tsx-mode` doesn't indent empty lines within `[]` and `()`, whereas empty lines within `{}` are indented. (`|` representing the cursor)

```js
[
|
]

(
|
)

{
  |
}
```
`js-mode` and `typescript-mode` appear to indent them all:
```js
[
  |
]

(
  |
)

{
  |
}
```

**Comment by AdamNiederer** [comment]
It might also be a good idea to check that the fix for this doesn't cause #30 for `[[` and `((`; `[[` is currently correctly indented to one stop.

**Comment by stevemolitor** [comment]
For the parenthesis case only, the issue seems to be with the [tree-sitter-typescript](https://github.com/tree-sitter/tree-sitter-typescript) grammar. It returns an `ERROR` node. Turning on `tree-sitter-debug-mode` on this TypeScript:

```ts
(

)
```

Produces this:

```
program:
  ERROR:
    formal_parameters:
```

I guess we could special case this and check for where the parent node is of type `ERROR` but that seems brittle. Better to fix the problem at the source.

Fwiw thing like this _do_ indent properly:

```
if (
  |
) {
  
}
```

I'm working on a fix for the `[` and `{` cases.
